symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/BdkConfigLoader.java

```diff
@@ -0,0 +1,156 @@
+
+@Slf4j
+public class BdkConfigLoader {
+
+    private static final JsonMapper JSON_MAPPER = new JsonMapper();
+    private static final YAMLMapper YAML_MAPPER = new YAMLMapper();
+
+    /**
+     * Load BdkConfig from a file path
+     *
+     * @param configPath Path of the config file
+     *
+     * @return Symphony Bot Configuration
+     */
+    public static BdkConfig loadFromFile(String configPath) {
+        try {
+            return JSON_MAPPER.readValue(new File(configPath), BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in JSON format");
+        }
+
+        try {
+            return YAML_MAPPER.readValue(new File(configPath), BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in Yaml format");
+        }
+
+        log.error("Config is not in a valid format");
+        return null;
+    }
+
+    /**
+     * Load BdkConfig from an InputStream
+     *
+     * @param inputStream InputStream
+     *
+     * @return Symphony Bot Configuration
+     */
+    public static BdkConfig loadFromInputStream(InputStream inputStream) {
+        try {
+            return JSON_MAPPER.readValue(inputStream, BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in JSON format");
+        }
+
+        try {
+            return YAML_MAPPER.readValue(inputStream, BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in Yaml format");
+        }
+
+        log.error("Config is not in a valid format");
+        return null;
+    }
+
+    /**
+     * Load BdkConfig from a classpath
+     *
+     * @param configPath Classpath to config file
+     *
+     * @return Symphony Bot Configuration
+     */
+    public static BdkConfig loadFromClasspath(String configPath) {
+        if (!configPath.startsWith(File.separator)) {
+            configPath = File.separator + configPath;
+        }
+        String externalUrlPath = System.getProperty("user.dir") + configPath;
+
+        BdkConfig config;
+        if ((new File(externalUrlPath)).exists()) {
+            config = loadFromFile(externalUrlPath);
+        } else {
+            InputStream is = BdkConfigLoader.class.getResourceAsStream(configPath);
+            if (is == null) {
+                is = Thread.currentThread().getContextClassLoader().getResourceAsStream(configPath);
+            }
+            config = loadFromInputStream(is);
+        }
+
+        return config;
+    }
+
+    /**
+     * Load LegacySymConfig from a config file
+     *
+     * @param configPath Path to config file
+     *
+     * @return Symphony Bot Legacy Configuration
+     */
+    public static LegacySymConfig loadLegacyConfigFromFile(String configPath) {
+        try {
+            return JSON_MAPPER.readValue(new File(configPath), LegacySymConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in JSON format");
+        }
```

Comment:
What about versioning the config file (with a version number in it) and have the same config object (wrapping the legacysymconfig in a bdkconfig)?

Reply:
Correct @symphony-youri, even when loading a legacy config file it should return a `BdkConfig` object. As well, I'd prefer to not have specific methods for legacy config loading, we should be able to detect it before mapping to target class : 

```
JsonNode config = parseConfigFile(path or inputstream or whatever)
if (config.at("/botUsername") != null) {
    // process legacy config format
} else {
    // new config format
}
```

As I also mentioned yesterday, we could also adopt the same strategy as used by `docker-compose.yml` file (cf. https://docs.docker.com/compose/) : 
```yaml
version: '2.0'
services:
  web:
    build: .
    ports:
    - "5000:5000"
    volumes:
    - .:/code
    - logvolume01:/var/log
    links:
    - redis
  redis:
    image: redis
volumes:
  logvolume01: {}
```
Where the version of the config format is part of the config file